### PR TITLE
TMEDIA 687 separator

### PR DESCRIPTION
**Pull request by JackHowa**
## Ticket

- [TMEDIA-687](https://arcpublishing.atlassian.net/browse/TMEDIA-687)

## Description

Takes in separator custom string via props. Set using ::after and scss per requirements 

## Acceptance Criteria

Provide a <Separator> component 

Default of dot separator 

Uses scss approach 

If a custom separator is provided, that is used

Have Scss file that adds the hooks to the Sass library

Have test coverage

Have storybook documentation

## Test Steps

1. Checkout branch - `git checkout TMEDIA-687-separator`
2. Run Storybook `npm run storybook`
3. Look for passing in separator string
4. Notice that the hook is still there for other customization than `content`
5. Check out the `separator` storybook documentation

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed relevant documentation has been updated/added.
- [x] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

### CODE REVIEW

File: src/components/separator/_index.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@use "../../scss";
+
+.c-separator {
+	@include scss.component-properties("separator");
+}
+
+.c-separator[data-separator-string]::after {
+	@include scss.component-properties("separator--after");
```
Comment (brentmiller1973) [inline src/components/separator/_index.scss]: @JackHowa , @matthewroach I'm a little confused as to why `seperator` and `separator--after` would be two separate entries in the advanced styling system.

File: src/components/separator/index.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Separator.propTypes = {
+	/** Class name(s) that get appended to default class name of the component */
+	className: PropTypes.string,
+	/** Custom separator string */
+	separatorString: PropTypes.string,
+};
+
+Separator.defaultProps = {
+	separatorString: "●",
```
Comment (matthewroach) [inline src/components/separator/index.jsx]: With the separator output being controlled via a React Prop means any themes supplied blocks that use the default separator are not able to be changed by a client via enhanced styling. If we move the separator value to be within the Sass we lift the control to theme settings

**Review by matthewroach — DISMISSED** [review]

File: src/components/separator/index.jsx (hunk 2 above)
Comment (JackHowa) [reply src/components/separator/index.jsx]: Oh ok. I must've misread the requirements that we wanted to pass in the separator string. can update to

```scss
@use "../../scss";

.c-separator {
	@include scss.component-properties("separator");
}

.c-separator::after {
	@include scss.component-properties("separator--after");
}
```

Is this what you're imagining?

Comment (matthewroach) [reply src/components/separator/index.jsx]: Allowing this prop is fine for when there is a need to have a different separator to the default

Comment (JackHowa) [reply src/components/separator/index.jsx]: gotcha, I think I follow. Updated

File: src/components/separator/index.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	const isCustomSeparator = separatorString !== "";
+	const classNames = [
+		COMPONENT_CLASS_NAME,
+		isCustomSeparator ? "c-separator--custom-string" : "",
+		className,
+	]
+		.filter((classString) => classString)
+		.join(" ");
+
+	if (isCustomSeparator) {
```
Comment (matthewroach) [inline src/components/separator/index.jsx]: I'd be curious if we can follow a similar pattern to how we tackled truncation on the paragraph component

File: src/components/separator/_index.scss (hunk 1 above)
Comment (matthewroach) [reply src/components/separator/_index.scss]: We should be able to achieve the enhanced styling hooks with one hook ```@include scss.component-properties("separator");```

**Comment by JackHowa** [comment]
hmm the variable generator doesn't seem to be respecting the string value. looking at it: 

theme.scss
```
separator: (
	content: "a",
),
```

-> 

Screen Shot 2022-03-14 at 11 19 37

should be

Screen Shot 2022-03-14 at 11 20 47

content property can only take in a string value

cc @matthewroach

**Review by matthewroach — APPROVED** [review]

**Pull request merged**